### PR TITLE
#160011184 Implement Like Dislike Articles 

### PR DESCRIPTION
**Pull request by janetnim**
#### What does this PR do?
Lets an authenticated user like or dislike an artice

#### Description of tasks to be completed
1. Allow authenticated users to like or dislike an article
2. Alert unauthenticated users to login/signup to like or dislike an article

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans-frontend.git
```
```sh
$ git checkout ft-like-dislike-articles-160011184
```
```sh
$ cd ah-titans-frontend
```

> Install the dependencies with the command below.

```sh
$ npm install
```
>Then run the app with the command below.
```sh
$ npm start
```
>Click on an article in the home page and like/dislike

### Relevant screenshots

>An article with no likes/dislikes
screen shot 2018-09-20 at 11 58 43

> Authenticated user likes/dislikes an article
screen shot 2018-09-20 at 11 59 13

> Unauthenticated user tries to like/dislike an article

screen shot 2018-09-20 at 12 02 34

#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#160011184](https://www.pivotaltracker.com/story/show/160011184)

### CODE REVIEW

**Comment by dcDalin** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :exclamation: Big PR
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Review by ronaldndirangu — APPROVED** [review]

**Review by migot01 — APPROVED** [review]

File: src/Home/components/index.js [hunk 1]
```diff
@@ -46,12 +46,14 @@ const ArticlesForm = ({ article, }) => (
...
         <p />
+      <Link to={`/article/${article.slug}`}>
         <p className="about">{article.description}</p>
         <br />
-        <p>
+        <p style={{color: "black"}}>
           {article.body && article.body.substring(0, 150)}
           {article.body && article.body.length > 150 ? '...' : ''}
         </p>
+        </Link>
```
Comment (jmnyarega) [inline src/Home/components/index.js]: Let's have this indented properly

File: src/Icons/components/icon.scss [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.icon_btn {
+    border: none;
+    background: none;
+}
```
Comment (jmnyarega) [inline src/Icons/components/icon.scss]: Add one more extra line at the end of this file

File: src/Icons/components/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import './icon.scss';
+
+const Icons = ({
+	id, icon, count, onClick,
+}) => (
+	<div>
+	<button className="icon_btn" onClick={onClick} >
+			<i className="material-icons" id={id}>{icon}</i>
+			<span>{count}</span>
+		</button>
```
Comment (jmnyarega) [inline src/Icons/components/index.js]: Let's have this indented properly

File: src/actions/likeDislike.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export const likeArticle = slug => (dispatch) => {
+	call({
+			endpoint: `/articles/${slug}/like/`,
+			method: 'PUT',
+		})
+		.then((data) => {dispatch(likeArticleActionCreator(data));
+        dispatch(likeDislikeArticleSuccess(data));
+			})
+		.catch(error => dispatch(likeDislikeErrorActionCreator(error)));
+};
```
Comment (jmnyarega) [inline src/actions/likeDislike.js]: Lets make use of the axios helper function her

**Review by jmnyarega — CHANGES_REQUESTED** [review]

**Comment by jmnyarega** [comment]
Please solve the conflicts and update your branch

**Comment by petermwash** [comment]
@jmnyarega Merge conflict solved and the branch is updated now.

**Pull request merged**